docs/arch/adr-008-weave-to-flannel-migration.md

```diff
@@ -0,0 +1,39 @@
+# ADR 8: Weave to Flannel Migration
+
+## Context
+
+As proposed in ADR 6, the future CNI of kURL will be Flannel.
+However, almost all existing installations are running Weave.
+
+A migration path must be provided for those instances.
+
+There were two general proposals for a migration path.
+One was to use Multus-CNI to allow using both Flannel and Weave at the same time before stripping out Weave, allowing an upgrade while only taking down one node at once.
+The other was to run a migration that removed Weave from all instances simultaneously before installing Flannel.
+This would be far simpler, not require an additional IP range, and not require Multus to remain in the cluster going forwards.
+It would also require 'stopping the world' and cause downtime.
+
+## Decision
+
+Provide a stop-the-world migration path to move from Weave to Flannel.
+
+## Solution
+
+Create a script to do the following:
+
+1. synchronize execution on all cluster nodes to minimize downtime
+2. Uninstall weave, removing iptables rules and host files
+3. Edit kubeadm.conf on all primary nodes and run kubeadm init
```

Comment:
Rather than running kubeadm init would it be better to follow the steps in this guide? https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-reconfigure/